### DISCUSSION TITLE
Setup behind azuracast nginx-proxy

### DISCUSSION THREAD

**Discussion by shogd3838** [oussamatn/radio-player#14] [Q&A] [answered]
Hello I have set up Azuracast behind nginx-proxy like described in the official documentation.
https://docs.azuracast.com/en/administration/docker/multi-site-installation

Now I want to add the radio-player application behind the same proxy but I can't succeed to make it work. 
Does someone here already succeed to do this type of installation ?

**Comment by shogd3838** [comment]
So finally I succeed to make the player work behind proxy with Azuracast on the same server.
I'll try to post a short documentation here when I have some free time while waiting for the offical one to be out.
Thank you oussamatn for your work. Much respect.

**Reply by burjuyz** [reply]
It would be great

**Reply by oussamatn** [reply]
Thank you for giving a try to make it work with the new Azuracast multi-site configuration. 
It will be great to share the solution with us. I'll write the documentation for it.

**Reply by shogd3838** [reply, marked as answer]
So to make the application works with Azuracast multi-site configuration once you have configured azuracast you need to create a second host in proxy server like this.

![image](https://user-images.githubusercontent.com/73856627/226404598-2df0224d-a3fe-4083-962b-7db344f51c0e.png)

Here the radio subdomain points to azuracast and player subdomain points to player PWA application. All containers azuracast, radio-player and nginx-proxy container run on the same server which local IP 192.168.1.10 is connected to WAN
Route your host "player" to port 8080 with simple http protocal and configure the proxy to handle SSL certificate and https.

![image](https://user-images.githubusercontent.com/73856627/226405763-244861f6-1151-4604-9f8b-8dfbc85ac2a0.png)
![image](https://user-images.githubusercontent.com/73856627/226405819-6021d358-11ee-419a-a9f7-19400f148937.png)

Once this is done you can build the radio-player image with only the radio-player exposing on port 8080. What I have done is to delete all docker compose files (main + override) and create a new docker-compose.yml with only this inside.

![image](https://user-images.githubusercontent.com/73856627/226409011-5d0c7e3f-dc41-4daa-bddf-9f74b7a8fffe.png)

**Reply by shogd3838** [reply]
One thing I forgot to tell is that you need to activate the "Use Web Proxy for Radio" option in azuracast or the audio streaming will not work behind nginx-proxy.
![image](https://user-images.githubusercontent.com/73856627/226593781-6b63b6cb-f5d6-4e5b-b49c-9048e4e986a7.png)

**Reply by xfiniteradiomedia** [reply]
This is very helpful thank you. I have the radio-player launched and it's running on the server the only issue I'm having is when I go to build the final product to launch it's throwing a front-end error. So for now what I decided to do is run "npm serve dist", it sends the build to a local server that I added to the proxy service and it still works just fine.

**Comment by xfiniteradiomedia** [comment]
Hello, I have successfully deployed the player to my hosting website but I am trying to deploy to firebase as a single page app. When I do this the loading symbol just spins. It works fine on my own hosting but I wanted to use it on firebase. Any thoughts.

**Reply by oussamatn** [reply]
Hi, I’m working on a small tutorial to explain how to deploy the player to free cloud hosting provider. I’ll post the link here later.

**Reply by xfiniteradiomedia** [reply]
Awesome thank you. I have setup the nginx proxy but still having an issue with the routing of the server. I look forward to your tutorial.

**Reply by xfiniteradiomedia** [reply]
Sorry to be a pain, I love the player its great. Would it be possible to add a feature and I also have a question. First is it possible to add a previous/next button down by the player to click to the next radio channels? Is there a way to add other radio stations that are not in my azuracast program. I'd be happy to send a donation to you if that's something you think is possible to add. Thanks so much again, great player!
